src/libs/auth.ts

```diff
@@ -105,15 +118,16 @@ export const signOut = (cause: SignOutCause = 'unspecified'): void => {
   azureCookieReadyStore.reset();
   getSessionStorage().clear();
   azurePreviewStore.set(false);
-  const auth = getAuthInstance();
+  const auth: AuthContextProps = getAuthInstance()!;
```

Comment:
this feels dangerous to not handle undefined auth case... any use of something! should come with a comment to explain why undefined case is being ignored... and here it feels like we should have hardened error handling concerns to at least throw an exception or log an error to help in debugging large auth flow problems.

Reply:
Agreed about the comment, but this seems fine. If `signOut` is being called, that implies that a user is authenticated.

Reply:
Then again... I'd suggest seeing if we can have `getAuthInstance` do the null check (https://github.com/DataBiosphere/terra-ui/pull/4249/files#r1331693833).

Reply:
Generally, that sounds perfectly reasonable.

In this case though, since `authInstance` is initialized so early, I'm wondering if adding the `| undefined` into the interface is add complexity to all consuming code for a case that's never practically encountered.